sympy/matrices/expressions/matexpr.py

```diff
@@ -1026,6 +1026,66 @@ def _entry(self, i, j, **kwargs):
         return S.One
 
 
+class MatrixWild(MatrixSymbol, Wild):
+    """A wildcard for matrix expressions based on ``Wild``.
+
+    Examples
+    ========
+
+    >>> from sympy import MatrixSymbol
+    >>> from sympy.matrices.expressions.matexpr import MatrixWild
+    >>> A, B = MatrixSymbol('A', 3, 3), MatrixSymbol('B', 3, 3)
+    >>> W = MatrixWild('W', 3, 3)
+    >>> (A*B).match(W)
+    {W_: A*B}
+    >>> (A*B).match(A*W)
+    {W_: B}
+    >>> (A**2).match(A*W)
+    {W_: A}
+
+    Note that ``W`` matched with ``A*B`` only because the expression had the
+    same shape as the matrix wildcard ``W``. If more flexibility is required, a
+    dimension of the matrix wildcard may itself be a wildcard.
+
+    >>> from sympy.core.symbol import Wild
+    >>> x = Wild('x')
+    >>> Y = MatrixWild('Y', x, 3)
+    >>> (A*B).match(Y)
+    {x_: 3, Y_: A*B}
+    """
+
+    def __new__(cls, name, n, m, exclude=(), properties=()):
+        obj = MatrixSymbol.__new__(cls, name, n, m)
+        obj.exclude = tuple([_sympify(x) for x in exclude])
+        obj.properties = tuple(properties)
+        return obj
+
+    def matches(self, expr, repl_dict={}, old=False):
```

Comment:
I know you use `copy` in the code, but it's safer to just use `repl_dict=None` here.

Reply:
I guess you have to copy it anyway, but seeing a mutable default value always makes me uneasy. What is the point of this argument?

Reply:
I see this is just copied from the core `matches`. It's rather bad style. The default really should be None everywhere, instead of `{}`.